src/Administration/Resources/app/administration/test/factory/component.factory.spec.js

```diff
@@ -846,4 +846,76 @@ describe('core/factory/component.factory.js', () => {
         expect(typeof component.vm.$super).toBe('function');
         expect(component.vm.$super('fooBar')).toBe('fooBar');
     });
+
+    it('builds the super call chain if first component in inheritance chain does not implement overridden method', () => {
+        ComponentFactory.register('grand-mother-component', {
+            template: '<div>This is a test template.</div>'
+        });
+        ComponentFactory.extend('mother-component', 'grand-mother-component', {
+            methods: {
+                fooBar() {
+                    return 'called';
+                }
+            }
+        });
+        ComponentFactory.extend('child-component', 'mother-component', {
+            methods: {
+                fooBar() {
+                    return this.$super('fooBar');
+                }
+            }
+        });
+
+        const childComponent = shallowMount(ComponentFactory.build('child-component'));
+
+        expect(childComponent.vm.fooBar()).toBe('called');
+    });
+
+    it('builds the super call chain if one component in inheritance chain does not implement overridden method', () => {
+        ComponentFactory.register('grand-mother-component', {
+            template: '<div>This is a test template.</div>',
+            methods: {
+                fooBar() {
+                    return 'called';
+                }
+            }
+        });
+        ComponentFactory.extend('mother-component', 'grand-mother-component', {});
+        ComponentFactory.extend('child-component', 'mother-component', {
+            methods: {
+                fooBar() {
+                    return this.$super('fooBar');
+                }
+            }
+        });
+
+        const childComponent = shallowMount(ComponentFactory.build('child-component'));
+
+        expect(childComponent.vm.fooBar()).toBe('called');
+    });
+
+    it('builds the super call chain if first components in inheritance chain do not implement overridden method', () => {
```

Comment:
The first component in the inheritance chain would bei `child-component`, so I'd use "root component" here instead.

```suggestion
    it('builds the super call chain correctly if the root component of the inheritance chain does not implement the overridden method', () => {
```

This one seems extraneous.